src/run/mod.rs

```diff
@@ -47,3 +58,55 @@ pub fn run(args: &RunArgs) -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// Determine which package should be run.
```

Comment:
Could you copy your comment from the PR into this function documentation?

> This PR adds a more sophisticated detection of the binary package detection.
> We first take a look at the `--bin` and `--package` args.
> If they are not defined, we try to determine the package automatically with the information provided by cargo metadata. We first look for the `default_run` definition and otherwise check if there is only a single binary package that could be run.

This will help others understand what this function is doing.

Reply:
Good idea, done!